scripts/generate_changelog.mjs

```diff
@@ -49,36 +52,96 @@ if (!fromTag) throw Error("No process.argv[2]");
 if (!toTag) throw Error("No process.argv[3]");
 if (!outpath) throw Error("No process.argv[4]");
 
+/**
+ * @type {Record<string, {heading: string; commitsByScope: Record<string, string[]>}>}
+ */
+const sections = {
+  feat: {heading: "Features", commitsByScope: {"": []}},
+  fix: {heading: "Bug Fixes", commitsByScope: {"": []}},
+  perf: {heading: "Performance", commitsByScope: {"": []}},
+  refactor: {heading: "Refactoring", commitsByScope: {"": []}},
+  revert: {heading: "Reverts", commitsByScope: {"": []}},
+  deps: {heading: "Dependencies", commitsByScope: {"": []}},
+  build: {heading: "Build System", commitsByScope: {"": []}},
+  ci: {heading: "Continuous Integration", commitsByScope: {"": []}},
+  test: {heading: "Tests", commitsByScope: {"": []}},
+  style: {heading: "Styles", commitsByScope: {"": []}},
+  chore: {heading: "Maintenance", commitsByScope: {"": []}},
+  docs: {heading: "Documentation", commitsByScope: {"": []}},
+  _: {heading: "Miscellaneous", commitsByScope: {"": []}},
+};
+
 const isPrCommitRg = /\(#\d+\)/;
+const conventionalCommitRg = /^([a-z]+)(?:\((.*)\))?(?:(!))?: (.*)$/;
```

Comment:
Thats pretty a pretty fancy regexp.  Did this get pulled from somewhere? Perhaps we can unit test it but that feels silly honestly for a workflow script... 

Found [this thread](https://gist.github.com/marcojahn/482410b728c31b221b70ea6d2c433f0c) and it had a a pretty cool one in it to support emojis.  

Not something I feel strongly about but thought was worth a mention.  I put my faith in your trusting hands sir

Reply:
Not in favor of adding tests for the script. I have verified that the output is correct.

As I also do not trust any regex, I created [some test cases](https://gist.github.com/nflaig/48ea28e05d292b0eeb1493dd4110ce48) which I just executed locally.

> a pretty cool one in it to support emojis

The regex allows emojis in the commit message itself. It is really not that strict, a more strict format will be enforced by the PR title checker.

Reply:
Added console logs to print out ignored commits (https://github.com/ChainSafe/lodestar/pull/5487/commits/dc26df1ebe591072237821c7c8517b8a86f8c190), this should be sufficient for now to verify that the regex works and there are no false positives.